chapter3.adoc

```diff
@@ -62,16 +40,50 @@ The field `relaxedpriv` in abstractcs (at 0x16) allows for relaxed permission ch
+System Bus Access enables direct reading/writing of memory space without involving the hart. However, it must always be checked by bus initiator protection mechanisms such as IOPMP, WorldGuard, etc. If these protections are not implemented or not deployed for Debug Module, System Bus Access must not be Supported. Failed system bus access attempts result in a bus security fault error (sberror 6).
+
+[NOTE]
+In scenarios where a Debug Module lacks System Bus Access, memory access by the debugger can be achieved through the use of abstract commands. These commands provide a secure means of accessing memory.
 
 [NOTE]
 Trusted entities like RoT should configure IOPMP or equivalent protection before granting debug access to machine mode. Similarly, machine mode should apply the protection before enabling submachine mode debug. 
 
 === Security Fault Error Reporting
 
-A dedicated error code, security fault error (cmderr 6), is included in cmderr of abstractcs (at 0x16) to signal any breaches of security enforcement by debug operations. Additionally, the bus security fault error (sberror 6) is introduced in sberror of sbcs (at 0x38) to denote errors related to system bus access.
+A dedicated error code, security fault error (cmderr 6), is included in cmderr of abstractcs (at 0x16) to signal any breaches of security enforcement by abstract commands. Additionally, the bus security fault error (sberror 6) is introduced in sberror of sbcs (at 0x38) to denote errors related to system bus access. The error raised by other debug operations which could be applied to multiple harts such as reset, keepalive can be determined by the fields allsecfault/anysecfault in dmstatus.
+
+
+=== Update of Debug Module Status (dmstatus, at 0x11)
+
+[caption="Register {counter:rimage}: ", reftext="Register {rimage}"]
+[title="The `allsecured` and `anysecured` fields in dmstatus"]
+[id=dmstatus]
+[wavedrom, ,svg]
+....
+{reg: [
+  {bits:   20, name: 'defined in Debug Module'},
+  {bits:   1, name: 'allsecured'},
+  {bits:   1, name: 'anysecured'},
+  {bits:   3, name: 'defined in Debug Module'},
+  {bits:   1, name: 'allsecfault'},
+  {bits:   1, name: 'anysecfault'},
+  {bits:   5, name: '0'},
+], config:{lanes: 3, hspace:1024}}
+....
+
+[[regdmstatus]]
+.The updates in dmstatus 
+[options="header"]
+|================================================================================================================================================
+| Field       | Description                                                                                                      | Access | Reset
+| allsecured  | The field is 1 when all currently selected harts are secured and not running in debuggable privilege levels      | R      | -    
```

Comment:
allsecured and anysecured fields are used for discovery purpose: whether this security extension is implemented or not. 
I propose to change its definition as follows: 

> The field is 1 when all currently selected harts implements Zedsec extension. 

Please also note that indicating the following condition in HW implementation is difficult to achieve: not running in debuggable privilege levels is

Reply:
I see. Sure, will update it.